### ISSUE TITLE
TGeoPainter build method doesn't fill name and other Object3d fields

### ISSUE THREAD

**Issue by DraTeots** [root-project/jsroot#303]
When using build() function to convert root geometry to Object3d, I noticed that nodes name fields are empty (see the screen below - red) and some other fields are not set properly (green). 

<img width="954" alt="image" src="https://github.com/root-project/jsroot/assets/4732545/3480ae7f-6656-4489-9aa3-8b582142f3cf">

I believe this is the reason, that if converters ( such as [root2cad](https://github.com/eic/root2cad) or [root to GLTF](https://github.com/HSF/root_cern-To_gltf-Exporter/blob/main/phoenixExport.js) ) produce geometry that is hard to navigate in consequent CAD or 3D software (see the screen, Blender is an example, others show the same): 

<img width="835" alt="image" src="https://github.com/root-project/jsroot/assets/4732545/04c2db16-5035-4836-a5f5-674b48945b6c">

Here is the reference geometry: 

[epic_dirc_only.root.zip](https://github.com/root-project/jsroot/files/15154172/epic_dirc_only.root.zip)

The code is straight forward: 

```typescript
const file = await openFile(url);
const geoManager = file.readObject('default');
console.time('Build geometry');
let obj3d = build(geoManager, { numfaces: 500000000, numnodes: 5000000, dflt_colors: true, vislevel: 10});
console.timeEnd('Build geometry');
console.log(obj3d);
```

**Comment by DraTeots** [comment]
A bit naive addition/suggestion (as I can imagine reasons it is not done). Besides names and hierarchy, there is [userData](https://threejs.org/docs/#api/en/core/Object3D.userData) field that can store e.g. TGeoNode reference in Object3D

**Comment by linev** [comment]
I see your point - all intermediate levels of geometry hierarchy did not get names.
And reason is simple - names were never used for these elements.
They are not "visible" and user cannot interact with them.

Why you need them?
May be you can use `stack` member which is assigned for each endpoint of hierarchy.
This `stack` member is somehow short form of path to the element.
One uses it for navigation, one also can produce full path to the element.

I also will check if `name` attribute can be assigned to the elements

**Comment by linev** [comment]
> Besides names and hierarchy, there is [userData](https://threejs.org/docs/#api/en/core/Object3D.userData) field that can store e.g. TGeoNode reference in Object3D

Main reason is saving JavaScript resources. 
Geometry can be large, therefore only minimal amount of data is assigned.
But I could extend `build()` function to assign more data to `Object3D`

**Comment by linev** [comment]
I add new parameters to [build()](https://jsroot.gsi.de/dev/jsdoc/global.html#build) function.

Now one can configure `opt.set_names` and `opt.set_origin`. First parameter is true by default.
Names and objects are assigned too all Object3D instances.

Code available in `master` branch only.

**Comment by DraTeots** [comment]
Thank you very much!

> Why you need them?

Actually in EIC they are used after GLTF export so even having node names in subsequent CAD and other software is good. 

I am in particular working on event display. Full EIC geometry in TGeo now has something like 3,500,000 elements and was working on scripts pruning the geometry and optimizing the rest after it is rendered with three.js. 

Thanks again for addressing this!

**Comment by DraTeots** [comment]
Would be very helpful if in InstancedGeometry mode some names were set to InstancedVolumes too. Maybe the volume name or the first node. Or maybe there is a mechanism already? As of now I see a list of InstancedGeometry-ies in the root of the Object3D without names.

**Comment by linev** [comment]
Now `name`, `names` and `userData` set also for `InstancedMesh` objects.
From the beginning also `stacks` was set.

**Comment by linev** [comment]
Now this functionality available in [7.7.0](https://github.com/root-project/jsroot/releases/tag/7.7.0) release
